### DISCUSSION TITLE
Version 3 API Transfer Learning not Updating Model

### DISCUSSION THREAD

**Discussion by rszabo15** [explosion/spaCy#7565] [Help: Coding & Implementations] [answered]
Hi - 
I am trying to do transfer learning in spacy version 3.0.0 using the en_core_web_trf and my own custom data set up in the format below using the API code. However, I have noticed that after running the training loop with the custom data, the model (nlp) does not actually update. All NER identifications on testing data and the meta.json file (except the disabled pipes) are both still exactly the same as the original en_core_web_trf model as if the training had no impact. In version 2, the code below (except converting the Examples object, added for version 3) worked as expected and the model was trained using the custom training data.
**How do I modify the code below to allow for the en_core_web_trf model to learn from the training data in version 3? Is there anything that I am missing?**

Note: I have tried to use the CLI training, but am having (separate) issues with that too. Please answer based on this current API code.

Training data (TRAIN_DATA):
[("Walmart is a leading e-commerce company", {"entities": [(0, 7, "ORG")]}),
("I reached Chennai yesterday.", {"entities": [(19, 28, "GPE")]})]

Code:
`
```
nlp = spacy.load('en_core_web_trf')
other_pipes = ['tagger', 'parser', 'attribute_ruler', 'lemmatizer'] # all but NER, transformer
nlp.disable_pipes(*other_pipes)

examples = []
for text, annotations in TRAIN_DATA:
    examples.append(Example.from_dict(nlp.make_doc(text), annotations))

optimizer = nlp.create_optimizer()
for itn in range(100):
    random.shuffle(examples)
    losses = {}
    batches = minibatch(examples, size=compounding(4,32,1.001)
    for batch in batches:
        nlp.update(batch, sgd=optimizer, drop=0.35,
                   losses=losses)

nlp.to_disk(<path>)
```
`

**Comment by svlandeg** [comment, marked as answer]
Hi @rszabo15, apologies for the late follow-up.

When running this example, note the warning that is printed:
```
UserWarning: [W030] Some entities could not be aligned in the text "I reached Chennai yesterday." with entities "[(19, 28, 'GPE')]". Use `spacy.training.offsets_to_biluo_tags(nlp.make_doc(text), entities)` to check the alignment. Misaligned entities ('-') will be ignored during training.
```
Which means that your second example contains entity offsets that can't be aligned - and thus will be ignored.

However, it does look like the `nlp` model is updating on the first (and only) example during the training loop, which becomes apparent when you print the losses (they fluctuate a little). That said, you might not be achieving the "overfitting" behaviour you were hoping to see. I think Matt's answer [here](https://github.com/explosion/spaCy/discussions/7942#discussioncomment-683818) is insightful in that regard.
